### PR TITLE
pg_up metric returns last calculated value without explicit resetting…

### PR DESCRIPTION
**Pull request by d-ulyanov**
If exporter is scraped by multiple Prometheuses (as we do) - Collect() could be called concurrently. In result in some cases one of Prometheuses could get pg_up = 0, because it was explicitly set to zero on first Collect call.

### CODE REVIEW

File: cmd/postgres_exporter/postgres_exporter.go [hunk 1]
```diff
@@ -812,21 +822,24 @@ func (s *Server) String() string {
...
-			log.Errorf("Error retrieving settings: %s", err)
-			errGauge.Inc()
+			err = fmt.Errorf("error retrieving settings: %s", err)
 		}
 	}
 
 	errMap := queryNamespaceMappings(ch, s)
 	if len(errMap) > 0 {
-		errGauge.Inc()
+		err = fmt.Errorf("queryNamespaceMappings returned %d errors")
```
Comment (vlamug) [inline cmd/postgres_exporter/postgres_exporter.go]: fmt.Errorf("queryNamespaceMappings returned %d errors", len(errMap))

File: cmd/postgres_exporter/postgres_exporter.go [hunk 2]
```diff
@@ -1280,16 +1293,42 @@ func (e *Exporter) scrape(ch chan<- prometheus.Metric) {
...
 	}
+
+	var errorsCount int
+	var hasConnectionErrors bool
+
 	for _, dsn := range dsns {
-		e.scrapeDSN(ch, dsn)
+		err := e.scrapeDSN(ch, dsn)
+
+		if err != nil {
```
Comment (vlamug) [inline cmd/postgres_exporter/postgres_exporter.go]: `if err := e.scrapeDSN(ch, dsn); err != nil {`

File: cmd/postgres_exporter/postgres_exporter.go [hunk 3]
```diff
@@ -1330,24 +1369,18 @@ func (e *Exporter) discoverDatabaseDSNs() []string {
...
 }
 
-func (e *Exporter) scrapeDSN(ch chan<- prometheus.Metric, dsn string) {
+func (e *Exporter) scrapeDSN(ch chan<- prometheus.Metric, dsn string) error {
 	server, err := e.servers.GetServer(dsn)
 	if err != nil {
-		log.Errorf("Error opening connection to database (%s): %v", loggableDSN(dsn), err)
-		e.error.Inc()
-		return
+		return &ErrorConnectToServer{fmt.Sprintf("Error opening connection to database (%s): %v", loggableDSN(dsn), err)}
```
Comment (vlamug) [inline cmd/postgres_exporter/postgres_exporter.go]: I think it is better to use %s placeholder for error

File: cmd/postgres_exporter/postgres_exporter.go (hunk 1 above)
Comment (d-ulyanov) [reply cmd/postgres_exporter/postgres_exporter.go]: thx

File: cmd/postgres_exporter/postgres_exporter.go (hunk 2 above)
Comment (d-ulyanov) [reply cmd/postgres_exporter/postgres_exporter.go]: okay)

**Comment by d-ulyanov** [comment]
@wrouesnel pls take a look for this issue

**Pull request merged**